src/bug.cpp

```diff
@@ -74,11 +105,18 @@ namespace simulation
     }
     auto free_adjacent = cell_->get_free_adjacent();
     if(!free_adjacent) {
+      breed_counter_--;
       return;
     }
-    std::shared_ptr<Bug> newborn_bug = std::make_shared<Bug>(free_adjacent);
+    breed_counter_ = 0;
+    std::shared_ptr<Bug> newborn_bug = std::make_shared<Bug>(free_adjacent, !is_prey_, rounds_to_breed_, rounds_to_starve_, name_ + ".");
```

Comment:
nice surname "." :rofl:

Reply:
Hahah tbh, It was useful to both occupy little space on the screen and still know who the parent the new bug was.